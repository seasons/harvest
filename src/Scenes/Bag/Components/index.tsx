export { BagTab } from "./BagTab"
export { ReservationHistoryTab } from "./ReservationHistoryTab"
export { SavedItemsTab } from "./SavedItemsTab"
