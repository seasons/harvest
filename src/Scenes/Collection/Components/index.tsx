export { ProductGrid } from "./ProductGrid"
export { CollectionText } from "./CollectionText"
