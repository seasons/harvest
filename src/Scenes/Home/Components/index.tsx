export { HomeBlogContent } from "./HomeBlogContent"
export { BrandsRail } from "./BrandsRail"
export { HomeFooter } from "./HomeFooter"
export { HomeBottomSheet } from "./HomeBottomSheet"
export { ProductsRail } from "./ProductsRail"
export { TagsRail } from "./TagsRail"
export { FitPicCollection } from "./FitPicCollection"
export { CategoriesRail } from "./CategoriesRail"
