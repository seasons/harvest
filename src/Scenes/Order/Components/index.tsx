export { ShippingOption } from "./ShippingOption"
export { SectionHeader } from "./SectionHeader"
