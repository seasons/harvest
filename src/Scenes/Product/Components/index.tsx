export { ImageRail } from "./ImageRail"
export { ProductDetails } from "./ProductDetails"
export { MoreLikeThis } from "./MoreLikeThis"
export { AboutTheBrand } from "./AboutTheBrand"
export { AddToBagButton } from "./AddToBagButton"
export { VariantWant } from "./VariantWant"
