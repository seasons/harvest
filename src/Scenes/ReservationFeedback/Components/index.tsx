export { ReservationFeedbackHeader } from "./ReservationFeedbackHeader"
export { ReservationFeedbackPopUp } from "./ReservationFeedbackPopUp"
export { ReservationFeedbackReminder } from "./ReservationFeedbackReminder"