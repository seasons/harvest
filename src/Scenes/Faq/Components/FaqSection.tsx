import React from "react"
import { Sans, Box, Separator, Spacer } from "App/Components"
import { color } from "App/utils"

export const FaqSection = ({ section }) => {
  return (
    <Box key={section.title}>
      <Box px={2}>
        <Sans size="3">{section.title}</Sans>
      </Box>
      <Spacer mb={3} />
      <Separator />
      <Box px={2}>
        {section.subsections.map(subsection => {
          return (
            <Box key={subsection.title}>
              <Spacer mb={3} />
              <Sans size="2">{subsection.title}</Sans>
              <Spacer mb={1} />
              <Sans size="2" color={color("black50")}>
                {subsection.text}
              </Sans>
            </Box>
          )
        })}
      </Box>
    </Box>
  )
}
