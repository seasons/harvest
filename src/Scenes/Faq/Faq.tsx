import React from "react"
import { Container, Spacer, FixedBackArrow, FixedButton, Flex } from "App/Components"
import gql from "graphql-tag"
import { useQuery } from "react-apollo"
import { Loader } from "App/Components/Loader"
import { FlatList, Linking } from "react-native"
import { FaqSection } from "./Components/FaqSection"

export const GET_FAQ = gql`
  query Faq {
    faq {
      sections {
        title
        subsections {
          title
          text
        }
      }
    }
  }
`

export const Faq = ({ navigation }) => {
  const { loading, error, data } = useQuery(GET_FAQ, {})

  if (loading) {
    return <Loader />
  }

  const renderItem = item => {
    return <FaqSection section={item} />
  }

  return (
    <Container insetsBottom={false}>
      <FixedBackArrow navigation={navigation} variant="whiteBackground" />
      <Flex style={{ flex: 1 }}>
        <FlatList
          ListHeaderComponent={() => <Spacer mb={80} />}
          data={data?.faq?.sections}
          keyExtractor={item => item.title}
          renderItem={({ item }) => renderItem(item)}
        />
      </Flex>
      <FixedButton
        block
        variant="primaryWhite"
        onPress={() => Linking.openURL(`mailto:membership@seasons.nyc?subject=Feedback`)}
      >
        Contact us
      </FixedButton>
    </Container>
  )
}
