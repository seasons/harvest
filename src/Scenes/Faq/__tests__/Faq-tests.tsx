import { waitForLoad } from "App/testUtils/waitForLoad"
import { mount } from "enzyme"
import React from "react"

import { MockedProvider } from "@apollo/client/testing"

import { FaqSection } from "../Components/FaqSection"
import { Faq, GET_FAQ } from "../Faq"

const FaqFixture = {
  faq: {
    sections: [
      {
        title: "Need to return your items?",
        subsections: [
          {
            title: "Ready to return your items?",
            text:
              "Pack up the pieces you want to return with the included hangers, attach the pre-paid shipping label and drop it off at your closest UPS or UPS pickup location.",
          },
          {
            title: "Did something not fit?",
            text: "We’re happy to help you find something that fits. Pack it up, send it back, and we’ll swap it out.",
          },
          {
            title: "Received the wrong item?",
            text:
              "Sorry about that! Pack it up, attach the pre-paid shipping label and send it back. We’ll get you the right one.",
          },
          {
            title: "Lost your return label?",
            text: "It happens. We understand. Contact us below and we’ll send you a new label.",
          },
        ],
      },
    ],
  },
}

describe("Faq", () => {
  const faqMocks = [
    {
      request: {
        query: GET_FAQ,
      },
      result: {
        data: {
          ...FaqFixture,
        },
      },
    },
  ]

  it("renders properly", async () => {
    const component = mount(
      <MockedProvider mocks={faqMocks} addTypename={false}>
        <Faq navigation={null} />
      </MockedProvider>
    )

    await waitForLoad(component)

    const faqSection = component.find(FaqSection)

    expect(faqSection.length).toEqual(1)
    expect(faqSection.text()).toContain(
      "Need to return your items?Ready to return your items?Pack up the pieces you want to return with the included hangers, attach the pre-paid shipping label and drop it off at your closest UPS or UPS pickup location.Did something not fit?We’re happy to help you find something that fits. Pack it up, send it back, and we’ll swap it out.Received the wrong item?Sorry about that! Pack it up, attach the pre-paid shipping label and send it back. We’ll get you the right one.Lost your return label?It happens. We understand. Contact us below and we’ll send you a new label."
    )
    expect(component).toMatchSnapshot()
  })
})
