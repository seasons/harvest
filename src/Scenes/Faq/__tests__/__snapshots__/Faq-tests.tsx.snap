// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Faq renders properly 1`] = `ReactWrapper {}`;
